### PR TITLE
Update polysolve and fix api changes

### PR DESCRIPTION
**Pull request by arvigj**

### CODE REVIEW

**Comment by Huangzizhou** [comment]
Could you update polysolve to 25ebacb10cc52fc6de13e9c24fc633ee7452fd50?

File: src/polyfem/solver/forms/Form.hpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <polyfem/utils/Types.hpp>
+#include <polysolve/nonlinear/Solver.hpp>
```
Comment (teseoch) [inline src/polyfem/solver/forms/Form.hpp:4]: include poststepdata

File: tests/verify_run.cpp [hunk 2]
```diff
@@ -122,7 +122,14 @@ int authenticate_json(const std::string &json_file, const bool compute_validatio
 	Eigen::MatrixXd sol;
 	Eigen::MatrixXd pressure;
 
-	state.solve_problem(sol, pressure);
```
Comment (teseoch) [inline tests/verify_run.cpp:125]: why this change?

Comment (Huangzizhou) [reply tests/verify_run.cpp:125]: @zfergus

Comment (zfergus) [reply tests/verify_run.cpp:125]: So it will continue testing the rest of the scenes and not crash when it runs out of iterations.

**Review by zfergus — APPROVED** [review]

File: src/polyfem/main.cpp [hunk 3]
```diff
@@ -215,7 +215,7 @@ int optimization_simulation(const CLI::App &command_line,
 	OptState opt_state;
 	opt_state.init(opt_args, is_strict);
 
-	opt_state.create_states(opt_state.args["compute_objective"].get<bool>() ? polyfem::solver::CacheLevel::Solution : polyfem::solver::CacheLevel::Derivatives);
+	opt_state.create_states(opt_state.args["compute_objective"].get<bool>() ? polyfem::solver::CacheLevel::Solution : polyfem::solver::CacheLevel::Derivatives, max_threads);
```
Comment (teseoch) [inline src/polyfem/main.cpp:218]: @Huangzizhou you changed this before, is it ok now?

Comment (Huangzizhou) [reply src/polyfem/main.cpp:218]: This is another issue that I didn't notice, should be fixed

**Review by Huangzizhou — APPROVED** [review]

File: src/polyfem/main.cpp (hunk 3 above)
Comment (arvigj) [reply src/polyfem/main.cpp:218]: So can I merge like this or do you want to fix something @Huangzizhou?

Comment (Huangzizhou) [reply src/polyfem/main.cpp:218]: oh wait this fix has a small issue, let me fix it
